analyzers/src/SonarAnalyzer.Common/Rules/BooleanLiteralUnnecessaryBase.cs

```diff
@@ -25,6 +25,8 @@ public abstract class BooleanLiteralUnnecessaryBase<TSyntaxKind> : SonarDiagnost
     {
         internal const string DiagnosticId = "S1125";
 
+        private INamedTypeSymbol[] systemBooleanInterfaces;
```

Comment:
```suggestion
        private ImmutableArray<INamedTypeSymbol> systemBooleanInterfaces;
```

Reply:
You may also consider:
* Loading the interfaces in CompilationStart and pass it from there or
* Use a static `ConditionalWeakTable<Compilation, ImmutableArray<INamedTypeSymbol>>` to store the interfaces between the instances of the analyzer.